game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/fire/FiringGroupTest.java

```diff
@@ -0,0 +1,187 @@
+package games.strategy.triplea.delegate.battle.steps.fire;
+
+import static games.strategy.triplea.Constants.UNIT_ATTACHMENT_NAME;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.mock;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.attachments.UnitAttachment;
+import java.util.Comparator;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class FiringGroupTest {
+
+  @Mock GameData gameData;
+  @Mock GamePlayer player;
+
+  @Test
+  public void onlyOneGroupIfNoSuicideOnHit() {
+    final List<Unit> units =
+        List.of(
+            givenUnitWithSuicideOnHit("type1", false), givenUnitWithSuicideOnHit("type2", false));
+    final List<FiringGroup> groups =
+        FiringGroup.groupBySuicideOnHit("test", units, List.of(mock(Unit.class)));
+    assertThat("There should only be one group", groups, hasSize(1));
+    assertThat("The group name should have no prefix", groups.get(0).getDisplayName(), is("test"));
+    assertThat(
+        "The group should have all of the units",
+        groups.get(0).getFiringUnits().toArray(),
+        is(units.toArray()));
+    assertThat("The group should not be suicide on hit", groups.get(0).isSuicideOnHit(), is(false));
+  }
+
+  private Unit givenUnitWithSuicideOnHit(final String typeName, final boolean suicideOnHit) {
```

Comment:
These two lines are pretty overwhelming to read, lots of "givenUnitWithSuicideOnHit", IMO the method names blur together and it becomes hard to tell exactly which API overload is being invoked where.

An intermediate variable could help, eg:

```
final var unitType = givenUnitTypeWithSuicideOnHit(typeName, suicideOnHit);
return givenUnitWithSuicideOnHit(unitType);
```